### PR TITLE
DD-1072: cleanup python scripts

### PR DESCRIPTION
**Pull request by lindareijnhoudt**
Fixes DD-1072

# Description of changes
* [x] dv-delete-draft-datasets
* [ ] update-datacite-records
* [ ] dv-convert-dataset-metadatafiles
* [x] dv-delete-role-assignment
* [x] dv-reindex-datasets
* [x] oai-harvest
* [x] dv-publish-datasets (first versions need to have type=major)
* [x] dv-replace-metadata-field-values
* [x] dv-retrieve-dataset-metadata
* [x] dv-retrieve-dtatset-pids
* [x] dv-unlock-datasets

done:
   * dv-replace-metadata-field-values only works for simple, non-repeating fields. Other options have not been tested extensively!
   * WIP replace_metadata_field_values
   * Check reindex_datasets, use similar argument names
   * Check delete_role_assignments
   * no defaults for required args
   * Check dv-delete-draft-datasets
   * Check convert_dataset_metadatafiles; TODO convert

# How to test

# Related PRs

(Add links)

*

# Notify

@DANS-KNAW/dataversedans

### CODE REVIEW

File: src/datastation/scripts/retrieve_dataset_pids.py [hunk 1]
```diff
@@ -6,23 +6,27 @@
...
+    store_pids(pids, output_dir, output_filename)
 
 
 def main():
     config = init()
 
     parser = argparse.ArgumentParser(
         description='Retrieves the pids for all published datasets in the given collection')
-    parser.add_argument('-o', '--output', help='The output file, for storing the pids retrieved')
+    parser.add_argument('-o', '--output-dir', default='.', dest='output_dir',
```
Comment (PaulBoon) [inline src/datastation/scripts/retrieve_dataset_pids.py:20]: output dir now defaults to '.', the working dir, but previously it was the configured dir now in 'files_root'. I have open an issue for this last config change (DD-1083). Maybe we can get rid of that issue and fix it here, if we want to?

File: src/datastation/scripts/unlock_datasets.py [hunk 2]
```diff
@@ -12,38 +12,30 @@
...
 
 
 def main():
     config = init()
     parser = argparse.ArgumentParser(description='Unlock datasets (if locked) with the pids in the given input file')
-    parser.add_argument('-p', '--pids-file', default='dataset_pids.txt', help='The input file with the dataset pids')
+    parser.add_argument('-i', '--input-file', dest='pids_file', help='The input file with the dataset pids')
     args = parser.parse_args()
-    unlock_dataset_command(config, args.pids_file)
+    unlock_dataset_command(config['dataverse']['server_url'], config['dataverse']['api_token'], args.pids_file)
```
Comment (PaulBoon) [inline src/datastation/scripts/unlock_datasets.py:38]: Indeed better to move these 'defaults' close to the input parsing.

**Comment by janvanmansum** [comment]
I did the following test on `dev_archaeology`: 

```
poetry run dv-replace-metadata-field-values -m citation -n title -f 'Manual Test' -t 'Changed title' -p 'doi:10.5072/FK2/PAGIFD'
2022-08-24 02:38:33 : DEBUG : init : Initialized logging
2022-08-24 02:38:34 : DEBUG : replace_metadata_field_value_action : doi:10.5072/FK2/PAGIFD: Found title with value Manual Test
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/janm/git/service/dataverse-dans/dans-datastation-tools/src/datastation/scripts/replace_metadata_field_values.py", line 102, in main
    replace_metadata_field_value_action(server_url, api_token, args.pid, args.mdb_name, args.field_name,
  File "/Users/janm/git/service/dataverse-dans/dans-datastation-tools/src/datastation/scripts/replace_metadata_field_values.py", line 52, in replace_metadata_field_value_action
    updated_field['value'][index_replace_from] = replace_to
TypeError: 'str' object does not support item assignment
```

**Review by janvanmansum — COMMENTED** [review]
* In the help strings make the option name and placeholder use the same term, e.g. instead of `-d PIDS` use `-d DATASETS`.

**Review by janvanmansum — COMMENTED** [review]
At some point, not necessarily now, we should probably create some naming conventions for the scripts, as there are going to be many of them. How about:
```
dv-<object>-<action>, e.g.,

dv-dataset-<action>, e.g.,
dv-dataset-publish
dv-dataset-replace-metadata-field

dv-user-<action>, e.g.,
dv-user-import
```
I am using the singular for the object assuming that every command supports operating both on a single object as well as on a list of objects.

File: src/datastation/scripts/oai_harvest.py [hunk 3]
```diff
@@ -51,8 +51,11 @@ def oai_harvest_command(server_url, output_dir, format, set=None):
 def main():
     config = init()
     parser = argparse.ArgumentParser(description='Harvest dataset metadata records via the OAI-PMH protocol')
-    parser.add_argument('-f', '--format', default='oai_dc', help='The format of the records to be harvested.')
-    parser.add_argument('-s', '--set', default='', help='The recordset to be harvested.')
+    parser.add_argument('-f', '--format', default='oai_dc',
+                        help='The PMH metadataPrefix in which the metadata should be harvested.')
+    parser.add_argument('-s', '--set', default='', help='The PMH recordset to be harvested.')
```
Comment (lindareijnhoudt) [inline src/datastation/scripts/oai_harvest.py:56]: it seems we haven't configured the sets in the Data Station

**Review by janvanmansum — CHANGES_REQUESTED** [review]
Remove `convert_dataset_metadatafiles.py`.

**Review by janvanmansum — CHANGES_REQUESTED** [review]
* Remove output_file argument from batch_process

File: src/datastation/scripts/oai_harvest.py [hunk 4]
```diff
@@ -22,14 +22,14 @@ def save_oai_records(xml_doc, counter, records_output_dir):
```
Comment (janvanmansum) [inline src/datastation/scripts/oai_harvest.py:23]: logging instead of prints

File: src/datastation/scripts/replace_metadata_field_values.py [hunk 5]
```diff
@@ -8,18 +10,17 @@
...
 
+def replace_metadatafield(server_url, api_token, pid, updated_field):
+    logging.debug("{}: Try updating it with: {}".format(pid, updated_field['value']))
+    updated_fields = {'fields': [updated_field]}
+    logging.debug(json.dumps(updated_fields))
+    replace_dataset_metadatafield(server_url, api_token, pid, updated_fields)
+    return True
+
+
 def replace_metadata_field_value_action(server_url, api_token, pid, mdb_name, field_name, field_from_value,
```
Comment (janvanmansum) [inline src/datastation/scripts/replace_metadata_field_values.py:21]:
```suggestion
def replace_metadata_field_value(server_url, api_token, pid, mdb_name, field_name, field_from_value,
```

File: src/datastation/scripts/replace_metadata_field_values.py [hunk 6]
```diff
@@ -74,26 +86,26 @@ def replace_metadata_field_value_command(config, server_url, api_token, pids_fil
...
-        description='Replace metadata field in datasets with the pids in the given inputfile')
-    parser.add_argument("metadatablock_name", help="Name of the metadata block")
-    parser.add_argument("field_name", help="Name of the field (json typeName)")
-    parser.add_argument("field_from_value", help="Value to be replaced")
-    parser.add_argument("field_to_value", help="Value replacing (the new value)")
-    parser.add_argument('-p', '--pids_file', default='dataset_pids.txt', help='The input file with the dataset pids')
+        description='Replace metadata field in datasets with the dois in the given input file. See the json metadata '
+                    'export (dataverse_json) to see what names are possible for the fields and metadata blocks. The '
+                    'field must have typeClass=`primitive`. The field must already be present.')
+    parser.add_argument("-m", "--metadata-block", help="Name of the metadata block", dest="mdb_name")
```
Comment (janvanmansum) [inline src/datastation/scripts/replace_metadata_field_values.py:92]: Remove? Not used?

**Review by janvanmansum — CHANGES_REQUESTED** [review]

**Pull request merged**